### PR TITLE
SCTUnit documentation

### PR DESCRIPTION
**Pull request by RainerKlute**

### CODE REVIEW

**Comment by RainerKlute** [comment]
Fixes issue #1395.

File: plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile [hunk 1]
```diff
@@ -5,370 +5,941 @@ p.
...
+The _exit_ statement exits and quits a state machine. You can re-initialize and re-enter it using the "_enter_ statement":#sctunit_entering_a_state_machine.
+
+
+==<div class="example">==
+
+Example:
+
+The statechart _myStatechart_ looks like this:
+
+!(tiny)images/sctunit_myStatechart_010_simple.png(Statechart _myStatechart_ )!
```
Comment (jthoene) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile]: image embedding is broken

File: plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile [hunk 2]
```diff
@@ -5,370 +5,941 @@ p.
...
+         * "State_A" should be active now: */
+        enter
+        assert active (myStatechart.main_region.State_A)
+    }
+}
+p. 
+
+==</div>==
+
+The _exit_ statement's grammar is simple, as figure "&quot;Exit statement grammar&quot;":#sctunit_fig_exit_grammar shows:
```
Comment (jthoene) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile]: If you view the documentation in the browser, this line is in the coding example.

Please fix this

File: plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile [hunk 3]
```diff
@@ -5,370 +5,941 @@ p.
...
+bc. var sum: integer = 0
+
+Figure "&quot;Variable definition statement grammar&quot;":#sctunit_fig_variable_definition_statement_grammar summarizes the structure of the _variable definition_ statement:
+
+p(#sctunit_fig_variable_definition_statement_grammar). 
+
+!images/sctunit_grammar_020_variabledefinitionstatement.png(Variable definition statement grammar)!
+
+p=. Variable definition statement grammar
+
```
Comment (jthoene) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile]: Maybe reference to the documentation of the Stext language (if one exists) and to VariableDefinition?

File: plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile [hunk 4]
```diff
@@ -5,370 +5,941 @@ p.
...
+        assert active(neutronFlux.main_region.State_A)
+        p = 12.0
+        proceed 1 cycle
+        assert active(neutronFlux.main_region.State_B)
+    }
+
+The effect of the two _mock_ statements is as follows:
+* If _getNeutronFlux(12.0)_ is called, the return value is 1000000000.0.
+* If _getNeutronFlux(18.0)_ is called, the return value is 4.3.
+* If _getNeutronFlux_ is called with any other parameter value, the return value is undefined.
```
Comment (jthoene) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile]: And the test will fail and throw an exception cause the return value and therefore the behaviour of the operation is undefined.

File: plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile [hunk 5]
```diff
@@ -5,370 +5,941 @@ p.
...
+
```
Comment (jthoene) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile]: Please add that if you don't specify the operation parameters value, the return value will be used for every operation call despite what value the operation parameter has.

File: plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile [hunk 6]
```diff
@@ -5,370 +5,941 @@ p.
...
+    TestClassB,
+    TestClassC
+}
+
+p. The test suite _MyTest_Suite_ comprises the test classes _TestClassA_, _TestClassB_, and _TestClassC_.
+
+==</div>==
+
+The nice thing about test suites is that you can run all tests of all the test classes at once. Right-click on the test suite file, say, _mytestsuite.sctunit_, and select _Run As → SCTUnit_ in the context menu. All tests in all test classes referenced in the test suite will be executed.
+
```
Comment (jthoene) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile]: Please add, that every referenced SCTUnit class in a SCTUnit suite must reference to statecharts with the same statechart type, ie if you have reference a SCTUnit class which reference to a CStatechart and another SCTUnit class which reference to a normal Default Statechart the SCTUnit Suite is not valid.

File: plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile [hunk 7]
```diff
@@ -5,370 +5,941 @@ p.
...
-Following example shows a minimal Test Group with namespace:
+A _test class_ contains one or more "operations":#sctunit_operations_and_tests or test cases. It is contained in a file with a _.sctunit_ filename extension.
 
-bc(sctunit). 
-testgroup MyExample for statechart SomeStatechart {
-	test firstTest{
-		enter		
-	}	
-}
+==<div class="example">==
```
Comment (RainerKlute) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile]: This is an issue with the GitHub viewer only. The generated HTML looks good.

File: plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile [hunk 8]
```diff
@@ -5,370 +5,941 @@ p.
...
+
+When running a "test class":#sctunit_test_classes or a "test suite":#sctunit_test_suites as an SCTUnit, only those operations are executed as tests that are annotated with <code>@Test</code>. Operations annotated with <code>@Ignore</code> or not annotated at all will not be regarded as tests. However, they can be called by test operations.
+
+While the <code>@Ignore</code> annotation and an omitted annotation have the same effect functionally, you should use the <code>@Ignore</code> annotation to mark operations that are intended as tests, but are (temporarily) disabled for one or the other reason. This differentiates them from other operations that are not tests, but mere subroutines called from elsewhere.
+
+
+
+h3(#sctunit_statements_and_expressions). Statements and expressions
+
+The body of an "operation":#sctunit_operations_and_tests consists of _statements_. Many types of statements, like _variable definitions_, _assignments_, _event raisings_, or _operation calls_, are already defined in the statechart language and are described "in section &quot;Statements&quot; of the statechart language documentation":../user-guide/statechart_language.html#statements.
```
Comment (RainerKlute) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile]: This is a GitHub bug. The generated HTML look fine.

File: plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile [hunk 9]
```diff
@@ -5,370 +5,941 @@ p.
...
+Generally, an assertion consists of the keyword @assert@, followed by a boolean expression, optionally followed by the keyword @message@ and an error message text. The assertion expects the boolean expression to be _true_ to continue the test. If it evaluates to _false_ the test fails. The optional message can be used to clarify what went wrong.
 
-p. ==<!-- End sctunit_keyword_assert -->==
 ==<!-- Start sctunit_keyword_called -->==
 
-h4. called
+h5. Asserting an operation call
 
-The expression followed by the 'called' keyword expects a reference to an operation that will be called and optional the keyword 'with' and the parameters of the call. If the referenced operation wasn't called or wasn't called with the right parameters, the test will fail. The following example shows a statement that asserts if a operation was called or not.
+A special variant of the assertion uses the _called_ expression. It checks whether a certain operation has been called (executed), typically by some action in the statechart.
```
Comment (RainerKlute) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile]: Hm, I rephrased the wording somewhat, but basically I'd like to keep it as it is.

The underlying general principle is that **people learn by three examples and the theory, not vice versa.**

File: plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile (hunk 1 above)
Comment (RainerKlute) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile]: This is also a GitHub viewer problem.

File: plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile (hunk 2 above)
Comment (RainerKlute) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile]: This is also a GitHub viewer problem.

File: plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile (hunk 3 above)
Comment (RainerKlute) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile]: It's already there, please see this section's introductory paragraph.

**Comment by RainerKlute** [comment]
Please recheck, @jthoene!

**Comment by jthoene** [comment]
There are still missing two changes. (s. line 774 and line 850)

**Comment by RainerKlute** [comment]
Don't know what you mean, @jthoene. I made these amendments as requested …

File: plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile [hunk 10]
```diff
@@ -5,370 +5,957 @@ p.
...
+* variables and operations defined on the test class level,
+* variables, operations, events, and states defined in an interface of the statechart controlled by the operation's test class,
+* entities imported by the statechart controlled by the operation's test class,
+* entities imported by the operation's test class.
+
+An operation can call other operations. This is like subroutine calls in other programming languages. Called operations can be …
+* located in the same test class,
+* located in "imported":#sctunit_the_import_statement test classes that pertain to the same statechart.
+
+Operations that are declared in the statechart _cannot_ be executed during a test. If you need them to be called you can "mocked":#sctunit_mocking_an_operation_call them.
```
Comment (RainerKlute) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile]: Fixed. Anything else?

**Pull request merged**